src/Results.Immutable.Tests/OptionTests/EqualityTests.cs

```diff
@@ -0,0 +1,58 @@
+using FsCheck;
+using FsCheck.Xunit;
+
+namespace Results.Immutable.Tests.OptionTests;
+
+public sealed class EqualityTests
+{
+    [Fact(DisplayName = "Equals to itself")]
+    public void EqualsToItself()
+    {
+        var option = Option.Some(1);
+        option.Should()
+            .Be(option);
+    }
+
+    [Property(DisplayName = "Equals to another object with the same value")]
+    public void EqualsToAnotherObjectWithTheSameValue() =>
+        Prop.ForAll<object?>(
+            static value =>
+            {
+                Option.Some(value)
+                    .Should()
+                    .Be(Option.Some(value));
+            });
+
+    [Property(DisplayName = "Not equals to another object with a different value")]
+    public void NotEqualsToAnotherObjectWithADifferentValue() =>
+        Prop.ForAll(
+            Arb.Generate<(object?, object?)>()
+                .Where(static value => value.Item1 != value.Item2)
+                .ToArbitrary(),
+            static value =>
+            {
+                Option.Some(value.Item1)
+                    .Should()
+                    .NotBe(Option.Some(value.Item2));
+            });
```

Comment:
- Stable release of FsCheck cannot generate value tuples; use `Tuple<T1, T2>` instead.
- Boxed structs will be compared by reference, making the `Where` predicate useless.

We can rewrite this test to a more robust property though:
```suggestion
[Property(DisplayName = "Performs proper equality comparison")]
public Property NotEqualsToAnotherObjectWithADifferentValue() =>
    Prop.ForAll(
        Arb.Generate<Tuple<object?, object?>>()
            .ToArbitrary(),
        static value =>
        {
            var (first, second) = value;

            return (first, second) switch
            {
                (not null, not null) when first.Equals(second) => Option.Some(first) ==
                    Option.Some(second),
                _ => Option.Some(first) != Option.Some(second),
            };
        });
```